src/cmd-cloud-prune

```
@@ -0,0 +1,303 @@
+    errors = []
+    totalBuildAMIs = len(build.images.get("amis") or "")
+    GCPImage = 0
+    # Unregister AMIs and snapshots
+    if not build.images.get("amis", []):
+        print(f"No AMIs for {build.id} for {build.arch}")
+    if not build.images.get("gcp", []):
+        print(f"No GCP image for {build.id} for {build.arch}")
+    else:
+        GCPImage = 1
+
+    if dry_run:
+        print(f"Would delete {totalBuildAMIs} amis/snapshots and {GCPImage} GCP image for {build.id} for {build.arch}")
+        return
+
+    for ami in build.images.get("amis", []):
+        print(f"Deleting cloud uploads for {build.id}")
+        region_name = ami.get("name")
+        ami_id = ami.get("hvm")
+        snapshot_id = ami.get("snapshot")
+        if ami_id and region_name:
+            try:
+                deregister_ami(ami_id, region=region_name)
+            except Exception as e:
+                errors.append(e)
+        if snapshot_id and region_name:
+            try:
+                delete_snapshot(snapshot_id, region=region_name)
+            except Exception as e:
+                errors.append(e)
+
+    gcp = build.images.get("gcp")
+    if gcp:
+        gcp_image = gcp.get("image")
+        json_key = gcp_cloud_config.get("gcp", {}).get("json-key")
+        project = gcp_cloud_config.get("gcp", {}).get("project")
+        if gcp_image and json_key and project:
+            try:
+                remove_gcp_image(gcp_image, json_key, project)
+            except Exception as e:
+                errors.append(e)
+
+    if len(errors) != 0:
+        print(f"Found errors when removing build {build.id}:")
+        for e in errors:
+            raise Exception(e)
+
+
+def cloud_uploads_check(actions):
+    if "cloud-uploads" in actions.keys():
+        cloud_uploads_duration = get_months(actions["cloud-uploads"])
+        build_duration = get_months(actions["build"])
+        assert cloud_uploads_duration < build_duration
```

```suggestion
        assert cloud_uploads_duration <= build_duration
```

---

Let's throw a proper exception instead with a clearer error message.